/* { dg-do run } */

defun foo = int:
  {
   try
     raise 1;
   catch (int e)
   {
     raise e;
   }
   return 666;
  }

defun bar = () int:
  {
   try
     return foo;
   catch
   {
     print ("caught exception\n");
   }

   return 0;
  }

/* { dg-command {print bar} } */
/* { dg-output "caught exception\n0" } */
